synapse/pipeline.yml

```
@@ -50,6 +50,17 @@ steps:
           image: "python:3.6"
           mount-buildkite-agent: false
 
+  - label: "\U0001F5A5 check unix line-endings"
+    command:
+      # Run 'file' command on each file, checking for Windows-style line terminators.
+      # grep returns exit code 1 when it doesn't find anything. We invert this as
+      # finding entries are a bad thing
+      - "find . -type f -path './.git/*' -prune -o -exec file '{}' ';' | (! grep 'CRLF line terminators')"
```

the trouble with this is that I'm not sure what makes `file` decide to print "CRLF line terminators", and I'd kinda like it to complain if there is a CRLF terminator anywhere in the file, even if they aren't used consistently throughout. I'd just use grep:

```suggestion
      - "find . -path './.git/*' -prune -o -type f -print0 | xargs -0 grep -l $'\r$' && { echo 'found files with CRLF line endings'; exit 1 }"
```

It might also be worth sticking that in a script in the synapse repo rather than here.

---

https://github.com/matrix-org/synapse/pull/7965